### DISCUSSION TITLE
Serial interface woes

### DISCUSSION THREAD

**Discussion by deavmi** [markqvist/Reticulum#620] [Help & Questions] [answered]
I have all the right permissions and have tested, using `picocom`, communication between the two devices and it works (I have a null-modem).

I have added the interface as per https://reticulum.network/manual/interfaces.html#serial-interface however when I start `rnsd` it says the serial interface is brought up yet i never see any traffic between my two `rnsd` daemons. Maybe I am doing something wrong, I just assumed placing both on the same link with interface configuration on either side would show something in the log, even when set to loglevel 7.

---

I am not doing anything yet between the nodes. I presume this may be the issue? Like I may need to generate some announcement that gets placed in an LXMF message and then routed as a reticulum packet through to the other node. I feel as though this may very well be the case. What is a good way to test then, perhaps?

**Comment by markqvist** [comment, marked as answer]
Reticulum is by default *very* quiet on configured physical links and interface, and will not send *anything* unless it absolutely needs to.

To see any traffic, the minimum thing you can do is send an announce from something. That should propagate out over any configured interface, and you should see it in the log of other device. You can use any existing application, or just fire up the `Announce.py` example and hit enter to send a couple of example announces.

**Reply by deavmi** [reply]
Perfect, going to give this a try.

**Reply by deavmi** [reply]
Will report back when done

**Reply by deavmi** [reply]
I did a few of these:

![image](https://github.com/user-attachments/assets/caaebb5c-7b89-4677-8ea4-e58b0a5a0b73)

Yet on the other node I don't see much:

![image](https://github.com/user-attachments/assets/deb9f81b-589b-4fe8-81d7-72a2affe1f5b)

**Reply by deavmi** [reply]
I take that back, I see logs after restart (`sen`, my docker CLI is retarded):

![image](https://github.com/user-attachments/assets/848c7cb9-0056-4767-bfef-19c9bd25657a)

**Comment by deavmi** [comment]
This is quite epic then. Thanks so far for the help!

I will come back to this soon, marked as good tho!

**Reply by deavmi** [reply]
Machine generating announces:

![image](https://github.com/user-attachments/assets/86987b6c-7de4-4d05-b700-bd6864a70d43)

Machine receiving it (over serial):

![image](https://github.com/user-attachments/assets/7ee436a0-bbad-4204-a647-75e1a80828f1)

**Reply by markqvist** [reply]
Very nice! Looks like it's all working then!

**Reply by deavmi** [reply]
Indeed, you don't know how happy I am about this :laughing:

**Reply by deavmi** [reply]
Also I know what was causing the lag, I think docker buffers stdout (because it saves logs to its own data store) and it only flushes to that data store after a large amount buffered. Hence why i would only "**see announces**" when I sent enough, when infact they were all there. Epic

**Reply by markqvist** [reply]
Ah, of course, I've run into that before with docker. As far as I remember, there is an option to flush at much lower intervals or immediately, but can't remember what is is :P
